com.unity.render-pipelines.universal/Runtime/UniversalGlobalSettings.cs

```diff
@@ -0,0 +1,179 @@
+                    var curGUID = guidHDGlobalAssets[0];
+                    path = AssetDatabase.GUIDToAssetPath(curGUID);
+                    assetCreated = AssetDatabase.LoadAssetAtPath<UniversalGlobalSettings>(path);
+                }
+                else if (canCreateNewAsset)// or create one altogether
+                {
+                    if (!AssetDatabase.IsValidFolder("Assets/" + folderPath))
+                        AssetDatabase.CreateFolder("Assets", folderPath);
+                    assetCreated = Create(path);
+
+                    Debug.LogWarning("No URP Global Settings Asset is assigned. One will be created for you. If you want to modify it, go to Project Settings > Graphics > URP Settings.");
+                }
+                else
+                {
+                    Debug.LogError("If you are building a Player, make sure to save an URP Global Settings asset by opening the project in the Editor first.");
+                    return null;
+                }
+            }
+            Debug.Assert(assetCreated, "Could not create URP's Global Settings - URP may not work correctly - Open  Project Settings > Graphics > URP Settings for additional help.");
+            UpdateGraphicsSettings(assetCreated);
+            return UniversalGlobalSettings.instance;
+        }
+
+        internal static UniversalGlobalSettings Create(string path, UniversalGlobalSettings src = null)
+        {
+            UniversalGlobalSettings assetCreated = null;
+
+            // make sure the asset does not already exists
+            assetCreated = AssetDatabase.LoadAssetAtPath<UniversalGlobalSettings>(path);
+            if (assetCreated == null)
+            {
+                assetCreated = ScriptableObject.CreateInstance<UniversalGlobalSettings>();
+                AssetDatabase.CreateAsset(assetCreated, path);
+                assetCreated.Init();
+                if (assetCreated != null)
+                {
+                    assetCreated.name = System.IO.Path.GetFileName(path);
+                }
+            }
+
+            if (assetCreated)
+            {
+                if (src != null)
+                {
+                    assetCreated.lightLayerName0 = System.String.Copy(src.lightLayerName0);
+                    assetCreated.lightLayerName1 = System.String.Copy(src.lightLayerName1);
+                    assetCreated.lightLayerName2 = System.String.Copy(src.lightLayerName2);
+                    assetCreated.lightLayerName3 = System.String.Copy(src.lightLayerName3);
+                    assetCreated.lightLayerName4 = System.String.Copy(src.lightLayerName4);
+                    assetCreated.lightLayerName5 = System.String.Copy(src.lightLayerName5);
+                    assetCreated.lightLayerName6 = System.String.Copy(src.lightLayerName6);
+                    assetCreated.lightLayerName7 = System.String.Copy(src.lightLayerName7);
+                }
+                else
+                {
+                }
```

Comment:
How do you initialize layer names in this case? Shouldn't you call ResetLayerNames here?